.github/workflows/validate_and_publish_release.yml

```diff
@@ -29,6 +29,7 @@ jobs:
         uses: actions/checkout@v2
         with:
           token: ${{ secrets.WRITE_ACCESS_TOKEN }}
+          fetch-depth: 200
```

Comment:
(following up on the thread at https://github.com/google/iree/pull/6932#issuecomment-911112205)

Fetching the `latest-snapshot` branch would be more natural, if that works.

The checkout action checks out the git reference used with the triggered workflow run and we want to push to the `latest-snapshot` branch:

```
* main
* other-commit-1
* other-commit-2
* new-snapshot
* other-commit-3
* other-commit-4 / latest-snapshot
* other-commit-5
```

The action runs at `new-snapshot`, but the push to `latest-snapshot` fails since `other-commit-3` and `other-commit-4` haven't been fetched. Increasing the fetch depth from `new-snapshot` would work as you have it, but that isn't expressing its intent directly and could flake if too many commits pass (unlikely).

Reply:
I agree we could add more CLIs in the checkout action as `run` to make the intent explicit, but it will still require increasing the checkout action fetch depth. 

```
- name: Checking out repository
        uses: actions/checkout@v2
  with:
        token: ${{ secrets.WRITE_ACCESS_TOKEN }}
        fetch-depth: 200
  run:
        git switch -c new-snapshot
        git fetch origin latest-snapshot:latest-snapshot
        git checkout latest-snapshot
        git merge new-snapshot
```

However, it would still fail at the same risk of not fetching enough depth from `new-snapshot`. If that's not a main concern, it seems we can skip the instructions in `run`.

If you think we should make it explicit, I am happy to make the change.

Reply:
Tested on my fork:
* https://github.com/ScottTodd/iree/runs/3497720330?check_suite_focus=true same error if no fetch depth or direct fetch
* https://github.com/ScottTodd/iree/runs/3497745982?check_suite_focus=true same error with direct fetch
* https://github.com/ScottTodd/iree/runs/3497800682?check_suite_focus=true success with force push

since we're treating this branch as a tag/label and don't care about actually merging changes or anything like that, let's just force push? (add `force: true` to the `github-push-action`'s args)

Reply:
(or see if Geoffrey has opinions - I'm against the arbitrary fetch-depth though :P)

Reply:
I think either force push @ push or fetch depth @ checkout (with a large depth) should work. I will modify the PR to use force push then.

Reply:
`fetch-depth: 0` will fetch all branches and tags and should guarantee to work, but it could also be an overkill for the simple notion of fastforwarding the latest-snapshot branch.

Reply:
Fetching all IREE history is nbd as long as we don't fetch submodules (https://github.com/google/iree/blob/bfd507f97a9076fce85021d7b2665630ef4f281e/.github/workflows/copybara_fixup.yml#L32-L34). I'd rather do that than have a magic number. I'd also prefer avoiding force push. It removes a safety that is pretty nice to keep. This should always be a fast forward. My vote would be to fetch the commit we want to push and then push it to the latest-snapshot branch (also, nit, I would drop `refs/heads` from the branch below, since it's unnecessary and people might be confused why it's used). I think all that's required to achieve this is to set this to `0`

Reply:
Okay I will make those two changes. Thanks for the feedback!